service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -525,7 +525,9 @@ private ApiCreateCloudContextResult fetchCreateCloudContextResult(
                     gcpCloudContextService.getRequiredGcpProject(workspaceId), userRequest),
             "enablePet");
     UserWithPetSa userAndPet = new UserWithPetSa(userEmail, petSaEmail);
```

Comment:
This is getting complex enough that I think we should move the logic out of the controller and into a service method. I'd suggest either the workspace service or the PetSaService.

In general, we don't want implementation in the controller; just dispatching.